test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,67 @@
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  - interval: 1m
```

Comment:
We only have 5 phases, so I think we can add the "regular" case for each. i.e. when the pod is not unschedulable, verify that each of the phases acts as expected - alerting for the 3 bad phases and not alerting for the good ones.